### PR TITLE
VersionBot: Append PR number of PR being merged to the `CHANGELOG.md` entry for it

### PR DESCRIPTION
**Pull request by hedss**
This allows tracing of each feature to its PR, and importantly will allow NotifyBot to
do the same to get back to Front conversations for informing customers of feature/fix
releases.

Connects-to: #230 

Change-type: minor
---- Autogenerated Waffleboard Connection: Connects to #230

### CODE REVIEW

File: lib/bots/versionbot.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ import { AlertLevel, LogLevel } from '../utils/logger';
 // Exec technically has a binding because of it's Node typings, but promisify doesn't include
 // the optional object (we need for CWD). So we need to special case it.
 const fsReadFile: (filename: string, options?: any) => Promise<Buffer | string> = Promise.promisify(FS.readFile);
+const fsWriteFile: (path: string, contents: string) => Promise<{}> = Promise.promisify(FS.writeFile);
```
Comment (CameronDiver) [inline lib/bots/versionbot.ts]: Is it worth using `mz` for these functions? They're already promisified and types are provided.

File: lib/bots/versionbot.ts [hunk 2]
```diff
@@ -1266,6 +1270,28 @@ export class VersionBot extends ProcBot {
...
 				versionData.version = match[1];
 				versionData.files = moddedFiles;
+
+				// Now we have to add the PR number and URL to every log entry we've just added for
+				// the new version. This ensures that NotifyBot can apply release notes later on, if
+				// required.
+				const versions = contents.split('## ');
+
+				// Now find the right version entry.
+				for (let index = 0; index < versions.length; index += 1) {
```
Comment (CameronDiver) [inline lib/bots/versionbot.ts]: Given that you only use `index` to retrieve the elements of the array, this should probably be a `for(let version of versions)`

File: lib/bots/versionbot.ts (hunk 1 above)
Comment (hedss) [inline lib/bots/versionbot.ts]: Having had a look, I'm going to stick with what I have atm due to the return of native Promises via `mz` and the mixing with Bluebird. I'd like to investigate what potential interactions should look like correctly. But it does look cleaner. :)

File: lib/bots/versionbot.ts (hunk 2 above)
Comment (hedss) [inline lib/bots/versionbot.ts]: Done.

Comment (CameronDiver) [inline lib/bots/versionbot.ts]: Ah yes, didn't notice that. Ignore me :)

**Comment by lekkas** [comment]
👍  all the pieces coming together :)

**Pull request merged**